src/transformStateWithClones.js

```diff
@@ -5,9 +5,47 @@
  * @param {Object[]} actions
  *
  * @return {Object[]}
+ *
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  /**  write code here
+  const stateHistory = [];
+  const copyState = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(copyState, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const element of action.keysToRemove) {
+        delete copyState[element];
+      }
+    } else if (action.type === 'clear') {
+      for (const key of Object.keys(copyState)) {
+        delete copyState[key];
+      }
+    }
+  */
+
+  const stateHistory = [];
+  const stateCopy = { ...state };
+
+  for (const action of actions) {
+    if (action.type === 'addProperties') {
+      Object.assign(stateCopy, action.extraData);
+    } else if (action.type === 'removeProperties') {
+      for (const element of action.keysToRemove) {
+        delete stateCopy[element];
+      }
+    } else if (action.type === 'clear') {
+      for (const key of Object.keys(stateCopy)) {
+        delete stateCopy[key];
+      }
+    }
```

Comment:
There is no default case in the if-else chain. In a switch statement, there should be a default case to handle unexpected action types.

Reply:
ok